### PR TITLE
Added 'Configuring Certificates' section OpenId README

### LINKED ISSUE #827: Orchard.OpenId access to signing cert's private key on Windows Server / IIS
I'm not 100% sure if this is the recommended action, but we needed to give Read access to the Private Key of the **signing certificate** to the AppPool Identity on Windows Server / IIS.

_Note that we started of with Full Control but have since reduced that to Read._

The cert was generated with PS `New-SelfSignedCertificate`.

If this is the recommended action, perhaps this is worth noting in the **README.md** for the Orchard.OpenId module.

Giving access to the AppPool Identity to the Private Key can be achieved in various ways, e.g.:

- MMC Certificate Snap-In (Right-Click on the Cert, All tasks > Manage private key > Add IIS AppPool\AppPoolName with Read access,
- winhttpcertcfg.exe 

Not doing so produced a 500 result and the following log entries:

~~~~
2017-06-15 07:56:35.0020|None|0HL5JOG5KGC5Q|59|Microsoft.Extensions.DependencyInjection.DataProtectionServices|WARN|Neither user profile nor HKLM registry available. Using an ephemeral key repository. Protected data will be unavailable when application exits. 
2017-06-15 07:56:35.1259|None|0HL5JOG5KGC5Q|50|Microsoft.AspNetCore.DataProtection.Repositories.EphemeralXmlRepository|WARN|Using an in-memory repository. Keys will not be persisted to storage. 
2017-06-15 07:56:37.7352|Default|0HL5JOG5KGC5Q|59|Microsoft.Extensions.DependencyInjection.DataProtectionServices|WARN|Neither user profile nor HKLM registry available. Using an ephemeral key repository. Protected data will be unavailable when application exits. 
2017-06-15 07:56:37.7523|Default|0HL5JOG5KGC5Q|50|Microsoft.AspNetCore.DataProtection.Repositories.EphemeralXmlRepository|WARN|Using an in-memory repository. Keys will not be persisted to storage. 
2017-06-15 07:56:38.8003|Default|0HL5JOG5KGC5Q|0|Orchard.OpenId.OpenIdConfiguration|ERROR|An error occurred while trying to register a X.509 certificate. 
2017-06-15 07:56:38.9227|Default|0HL5JOG5KGC5Q|13|Microsoft.AspNetCore.Server.Kestrel|ERROR|Connection id "0HL5JOG57HEGH": An unhandled exception was thrown by the application. Internal.Cryptography.CryptoThrowHelper+WindowsCryptographicException: The system cannot find the file specified
   at System.Security.Cryptography.CngKey.Open(String keyName, CngProvider provider, CngKeyOpenOptions openOptions)
   at System.Security.Cryptography.CngKey.Open(String keyName, CngProvider provider)
   at Internal.Cryptography.Pal.CertificatePal.GetPrivateKey[T](Func`2 createCsp, Func`2 createCng)
   at Internal.Cryptography.Pal.CertificatePal.GetRSAPrivateKey()
   at Internal.Cryptography.Pal.CertificateExtensionsCommon.GetPrivateKey[T](X509Certificate2 certificate, Predicate`1 matchesConstraints)
   at Microsoft.IdentityModel.Tokens.X509SecurityKey.get_PrivateKey()
   at Microsoft.IdentityModel.Tokens.X509SecurityKey.get_HasPrivateKey()
   at Microsoft.AspNetCore.Builder.OpenIdConnectServerExtensions.AddKey(IList`1 credentials, SecurityKey key)
   at Microsoft.AspNetCore.Builder.OpenIdConnectServerExtensions.AddCertificate(IList`1 credentials, X509Certificate2 certificate)
   at Microsoft.AspNetCore.Builder.OpenIdConnectServerExtensions.AddCertificate(IList`1 credentials, String thumbprint, StoreName name, StoreLocation location)
   at Orchard.OpenId.OpenIdConfiguration.Configure(OpenIddictOptions options) in D:\Repos\GH\zumeymedia\Orchard2\src\OrchardCore.Modules\Orchard.OpenId\Services\OpenIdConfiguration.cs:line 54
   at Microsoft.Extensions.Options.OptionsCache`1.CreateOptions()
   at System.Threading.LazyInitializer.EnsureInitializedCore[T](T& target, Boolean& initialized, Object& syncLock, Func`1 valueFactory)
   at Microsoft.Extensions.Options.OptionsCache`1.get_Value()
   at Microsoft.AspNetCore.Builder.OpenIddictExtensions.UseOpenIddict(IApplicationBuilder app)
   at Orchard.OpenId.Startup.Configure(IApplicationBuilder builder, IRouteBuilder routes, IServiceProvider serviceProvider) in D:\Repos\GH\zumeymedia\Orchard2\src\OrchardCore.Modules\Orchard.OpenId\Startup.cs:line 54
   at Microsoft.AspNetCore.Modules.ModularTenantRouterMiddleware.BuildTenantPipeline(ShellSettings shellSettings, IServiceProvider serviceProvider) in D:\Repos\GH\zumeymedia\Orchard2\src\OrchardCore\Microsoft.AspNetCore.Modules\ModularTenantRouterMiddleware.cs:line 105
   at Microsoft.AspNetCore.Modules.ModularTenantRouterMiddleware.<Invoke>d__4.MoveNext() in D:\Repos\GH\zumeymedia\Orchard2\src\OrchardCore\Microsoft.AspNetCore.Modules\ModularTenantRouterMiddleware.cs:line 72
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
   at Microsoft.AspNetCore.Modules.ModularTenantContainerMiddleware.<Invoke>d__4.MoveNext() in D:\Repos\GH\zumeymedia\Orchard2\src\OrchardCore\Microsoft.AspNetCore.Modules\ModularTenantContainerMiddleware.cs:line 77
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.AspNetCore.Server.IISIntegration.IISMiddleware.<Invoke>d__8.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.AspNetCore.Hosting.Internal.RequestServicesContainerMiddleware.<Invoke>d__3.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Http.Frame`1.<RequestProcessingAsync>d__2.MoveNext()    at System.Security.Cryptography.CngKey.Open(String keyName, CngProvider provider, CngKeyOpenOptions openOptions)
   at System.Security.Cryptography.CngKey.Open(String keyName, CngProvider provider)
   at Internal.Cryptography.Pal.CertificatePal.GetPrivateKey[T](Func`2 createCsp, Func`2 createCng)
   at Internal.Cryptography.Pal.CertificatePal.GetRSAPrivateKey()
   at Internal.Cryptography.Pal.CertificateExtensionsCommon.GetPrivateKey[T](X509Certificate2 certificate, Predicate`1 matchesConstraints)
   at Microsoft.IdentityModel.Tokens.X509SecurityKey.get_PrivateKey()
   at Microsoft.IdentityModel.Tokens.X509SecurityKey.get_HasPrivateKey()
   at Microsoft.AspNetCore.Builder.OpenIdConnectServerExtensions.AddKey(IList`1 credentials, SecurityKey key)
   at Microsoft.AspNetCore.Builder.OpenIdConnectServerExtensions.AddCertificate(IList`1 credentials, X509Certificate2 certificate)
   at Microsoft.AspNetCore.Builder.OpenIdConnectServerExtensions.AddCertificate(IList`1 credentials, String thumbprint, StoreName name, StoreLocation location)
   at Orchard.OpenId.OpenIdConfiguration.Configure(OpenIddictOptions options) in D:\Repos\GH\zumeymedia\Orchard2\src\OrchardCore.Modules\Orchard.OpenId\Services\OpenIdConfiguration.cs:line 54
   at Microsoft.Extensions.Options.OptionsCache`1.CreateOptions()
   at System.Threading.LazyInitializer.EnsureInitializedCore[T](T& target, Boolean& initialized, Object& syncLock, Func`1 valueFactory)
   at Microsoft.Extensions.Options.OptionsCache`1.get_Value()
   at Microsoft.AspNetCore.Builder.OpenIddictExtensions.UseOpenIddict(IApplicationBuilder app)
   at Orchard.OpenId.Startup.Configure(IApplicationBuilder builder, IRouteBuilder routes, IServiceProvider serviceProvider) in D:\Repos\GH\zumeymedia\Orchard2\src\OrchardCore.Modules\Orchard.OpenId\Startup.cs:line 54
   at Microsoft.AspNetCore.Modules.ModularTenantRouterMiddleware.BuildTenantPipeline(ShellSettings shellSettings, IServiceProvider serviceProvider) in D:\Repos\GH\zumeymedia\Orchard2\src\OrchardCore\Microsoft.AspNetCore.Modules\ModularTenantRouterMiddleware.cs:line 105
   at Microsoft.AspNetCore.Modules.ModularTenantRouterMiddleware.<Invoke>d__4.MoveNext() in D:\Repos\GH\zumeymedia\Orchard2\src\OrchardCore\Microsoft.AspNetCore.Modules\ModularTenantRouterMiddleware.cs:line 72
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
   at Microsoft.AspNetCore.Modules.ModularTenantContainerMiddleware.<Invoke>d__4.MoveNext() in D:\Repos\GH\zumeymedia\Orchard2\src\OrchardCore\Microsoft.AspNetCore.Modules\ModularTenantContainerMiddleware.cs:line 77
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.AspNetCore.Server.IISIntegration.IISMiddleware.<Invoke>d__8.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.AspNetCore.Hosting.Internal.RequestServicesContainerMiddleware.<Invoke>d__3.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Http.Frame`1.<RequestProcessingAsync>d__2.MoveNext()
2017-06-15 07:57:18.5309|Default|0HL5JOG5KGC5R|35|Microsoft.AspNetCore.DataProtection.KeyManagement.XmlKeyManager|WARN|No XML encryptor configured. Key {c2467ae9-ec5a-4799-a3dc-5aeea0cf9ad9} may be persisted to storage in unencrypted form. 
2017-06-15 07:57:22.9585|Default|0HL5JOG5KGC5S|35|Microsoft.AspNetCore.DataProtection.KeyManagement.XmlKeyManager|WARN|No XML encryptor configured. Key {36399eae-7155-4184-a16c-aee75b7035ac} may be persisted to storage in unencrypted form. 
2017-06-15 07:57:22.9872|Default|0HL5JOG5KGC5S|7|Microsoft.AspNetCore.Antiforgery.Internal.DefaultAntiforgery|ERROR|An exception was thrown while deserializing the token. System.InvalidOperationException: The antiforgery token could not be decrypted. ---> System.Security.Cryptography.CryptographicException: The key {cb5a586d-969e-4e61-884a-259677f15430} was not found in the key ring.
   at Microsoft.AspNetCore.DataProtection.KeyManagement.KeyRingBasedDataProtector.UnprotectCore(Byte[] protectedData, Boolean allowOperationsOnRevokedKeys, UnprotectStatus& status)
   at Microsoft.AspNetCore.DataProtection.KeyManagement.KeyRingBasedDataProtector.DangerousUnprotect(Byte[] protectedData, Boolean ignoreRevocationErrors, Boolean& requiresMigration, Boolean& wasRevoked)
   at Microsoft.AspNetCore.DataProtection.KeyManagement.KeyRingBasedDataProtector.Unprotect(Byte[] protectedData)
   at Microsoft.AspNetCore.Antiforgery.Internal.DefaultAntiforgeryTokenSerializer.Deserialize(String serializedToken)
   --- End of inner exception stack trace ---
   at Microsoft.AspNetCore.Antiforgery.Internal.DefaultAntiforgeryTokenSerializer.Deserialize(String serializedToken)
   at Microsoft.AspNetCore.Antiforgery.Internal.DefaultAntiforgery.GetCookieTokenDoesNotThrow(HttpContext httpContext)    at Microsoft.AspNetCore.Antiforgery.Internal.DefaultAntiforgeryTokenSerializer.Deserialize(String serializedToken)
   at Microsoft.AspNetCore.Antiforgery.Internal.DefaultAntiforgery.GetCookieTokenDoesNotThrow(HttpContext httpContext)
~~~~

### PR DESCRIPTION
**Pull request by anoordende**
Resolves #827

### CODE REVIEW

**Comment by jersiovic** [comment]
Thank you @anoordende for taking your time for improving documentation

File: src/OrchardCore.Modules/Orchard.OpenId/README.md [hunk 1]
```diff
@@ -103,3 +103,25 @@ OpenID Connect apps require the following configuration.
       "AllowHybridFlow": false
 }
 ```
+
+## Configuring Certificates
+
+### Windows / IIS
+
+Several tools are availble for generating a signing certificate on Windows and/or IIS, for example:
```
Comment (PinpointTownes) [inline src/OrchardCore.Modules/Orchard.OpenId/README.md]: nit: `availble` -> `available` :smile:

File: src/OrchardCore.Modules/Orchard.OpenId/README.md [hunk 2]
```diff
@@ -103,3 +103,25 @@ OpenID Connect apps require the following configuration.
...
+
++ IIS Server Manager _(offers not much control)_
+    1. Server Certificates
+    2. Create Self-Signed Certificate 
++ PowerShell _(offers full control)_
+    1. `New-SelfSignedCertificate` https://technet.microsoft.com/en-us/itpro/powershell/windows/pkiclient/new-selfsignedcertificate
```
Comment (PinpointTownes) [inline src/OrchardCore.Modules/Orchard.OpenId/README.md]: This cmdlet is insanely obscure/complicated for beginners, so we should consider giving a concrete example. E.g:

```
New-SelfSignedCertificate `
    -Subject "connect.example.com" `
    -FriendlyName "Example.com Signing Certificate" `
    -CertStoreLocation "cert:\LocalMachine\My" `
    -KeySpec Signature `
    -KeyUsage DigitalSignature `
    -KeyUsageProperty Sign `
    -TextExtension @("2.5.29.37={text}1.3.6.1.5.5.7.3.1") `
    -KeyExportPolicy NonExportable `
    -KeyAlgorithm RSA `
    -KeyLength 4096 `
    -HashAlgorithm SHA256 `
    -NotAfter (Get-Date).AddYears(5) `
    -Provider "Microsoft Enhanced RSA and AES Cryptographic Provider"
```

This snippet must run as admin. It generates a 4096-bit signing certificate, stores it in the machine store and returns the thumbprint that must be set in the OpenID module settings.

(not tested, so please give it a try before using it as-is :sweat_smile:)

**Comment by PinpointTownes** [comment]
@anoordende thanks for your PR! I just have a few nits, but it looks fine :+1:

File: src/OrchardCore.Modules/Orchard.OpenId/README.md (hunk 1 above)
Comment (anoordende) [inline src/OrchardCore.Modules/Orchard.OpenId/README.md]: Crap, no spellcheck!

File: src/OrchardCore.Modules/Orchard.OpenId/README.md (hunk 2 above)
Comment (anoordende) [inline src/OrchardCore.Modules/Orchard.OpenId/README.md]: True, I'll test it and update it. Good point it has to be NonExportable.

Comment (PinpointTownes) [inline src/OrchardCore.Modules/Orchard.OpenId/README.md]: Hum, maybe we should remove `-KeyExportPolicy NonExportable` for a "public"/general example. Folks who decide to deploy their Orchard app to multiple servers might have a bad surprise :sweat_smile: 

(that said, even `NonExportable` certificates *can* be exported, using tools like `mimikatz` or https://github.com/iSECPartners/jailbreak, it's just a bit more complicated :sweat_smile:)

Comment (anoordende) [inline src/OrchardCore.Modules/Orchard.OpenId/README.md]: Generation tested OK on Win10 and WinSrv2016, will try and use it too. This is undoubtedly going to be copied & pasted without hesitation, so should we reduce the 5 years, perhaps to 825 days? As per the (equally obscure-) new SSL/TLS validity guidelines.

Comment (PinpointTownes) [inline src/OrchardCore.Modules/Orchard.OpenId/README.md]: Sounds like a good idea (even if the signing certificate is not used for SSL/TLS).

It reminds me I need to add a check in the OIDC server middleware to throw an exception when trying to register an outdated certificate.

Comment (anoordende) [inline src/OrchardCore.Modules/Orchard.OpenId/README.md]: I had some issues with exportable certificates, but let me revisit what it was tomorrow as it's Friday Beer O'Clock now.

**Comment by anoordende** [comment]
Incorporated reviews and expanded on the Exportable/Non-Exportable. All script / commandline examples tested.

**Comment by sebastienros** [comment]
@PinpointTownes Are you satisfied with the changes?

**Comment by PinpointTownes** [comment]
@sebastienros I am!

@anoordende thanks for your contribution :+1:

**Comment by Jetski5822** [comment]
boom!

**Pull request merged**